Feat: Onboarding for first time users

MarcK0909 opened the pull request:
## Summary
Replace the beta consent screen with a new onboarding scren and wire it into post-registration and login flows.  
Align navigation, flags, and tests with the onboarding flow; add illustrative assets for the tutorial slides.  
Refresh end-to-end utilities to skip onboarding where needed. This might be further built upon to allow the user to redo the onboarding by clicking a button from his account in a future PR.

## What

### UI
- Rebuilt the beta consent screen into an **onboarding pager** with four feature slides, skip/next controls, and embedded images (`OnboardingScreen`).
- Added new drawable assets for onboarding slides (`star_items.png`, `find_friends.png`).

### ViewModel
- Simplified onboarding completion tracking to a local **SharedPreferences flag** (`OnboardingViewModel`); removed consent/Firebase dependency.
- Updated navigation to use `Screen.Onboarding` after registration and mark onboarding as seen on skip/done.

### Tests
- Updated onboarding UI tests (`OnboardingScreenTest`) and registration helper to use new tags/flow.
- Updated end-to-end setup to pre-set the onboarding flag (Firestore-based tests) to skip onboarding in automated flows.
- Refreshed onboarding ViewModel unit/factory tests to match the new local-flag logic.

## Why
First-run experience now teaches core features instead of asking for beta consent, matching product goals for new users.  
Using a local onboarding flag removes unnecessary network dependency and simplifies both flows and testing.

## Screenshots / Videos
New onboarding carousel with feature slides (see attached assets `star_items.png`, `find_friends.png` in the app).
Below is the preview mode of this new feature as going to the real screen was tedious.


https://github.com/user-attachments/assets/9530b1c1-4e76-4ae8-af49-4d47b98b9d52


## Checklist
- [x] Tests: added/updated (onboarding UI tests, ViewModel unit tests, e2e setup adjustments)  
- [ ] Documentation: n/a  
- [x] Manual verification: ran onboarding flow after registration; confirmed skip/done navigates to feed and assets render on slides

cocoStein commented on app/src/androidTest/java/com/android/ootd/end2end/fourthEnd2endTest.kt:
```diff
@@ -108,6 +113,7 @@ class FourthEnd2EndTest : FirestoreTest() {
     FirebaseEmulator.auth.signOut()
 
     initTestNavController()
+    Log.d(TAG, "Setting content and launching OOTDApp")
```
That’s a great idea to put some logs to test the file, but I don’t know if I really like to publish them to prod, as they are used to debug. 

It will create too many unnecessary logs if the test works. Either we could make a `debug mode` where the logs are on, and a normal mode (default) where they are not. Or we could just remove them.

Let me know what you think

cocoStein commented on app/src/main/java/com/android/ootd/ui/onboarding/OnboardingScreen.kt:
```diff
@@ -0,0 +1,287 @@
+  const val PAGE_TITLE_PREFIX = "onboardingPageTitle_"
+}
+
+data class OnboardingPage(
```
Small question, but why did you call this `Onboarding`, but your PR is called Tutorial? It's very nitpicky, but I found it also clearer if it were called `Tutorial`.

Let me know what you think

MarcK0909 replied:
I feel like tutorial sounds more like a video game wheras onboarding is basically just explaining the app to first-time users.

cocoStein replied:
Ok ok if you see it this way

MarcK0909 commented on app/src/main/java/com/android/ootd/ui/onboarding/OnboardingScreen.kt:
```diff
@@ -0,0 +1,287 @@
+                    onClick = onSkip,
+                    enabled = !isLoading,
+                    modifier = Modifier.testTag(OnboardingScreenTestTags.SKIP_BUTTON)) {
+                      Text("Skip", style = Typography.labelLarge, color = OnSurfaceVariant)
```
Yes I added a background color for more visibility.

MarcK0909 commented:
> Hey great PR and it will really improve the First time User experience.
> 
> I just added some small comments about the text and also all the colors you used are wrong, you used the default one of the device and not the one in the `Colors.kt`. So for example on my emulator it looks totally different than you
> 
> <img alt="Image" width="540" height="1200" src="https://private-user-images.githubusercontent.com/79094799/525992049-2b1ba6e4-d4c4-4766-ad4c-a096c02f6a47.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU1NjgxMjUsIm5iZiI6MTc2NTU2NzgyNSwicGF0aCI6Ii83OTA5NDc5OS81MjU5OTIwNDktMmIxYmE2ZTQtZDRjNC00NzY2LWFkNGMtYTA5NmMwMmY2YTQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjEyVDE5MzAyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0MDIyNjMyNTgxODk3ZmIxN2RhMTVmMDVjMDlhMDg3YWI5MzZiMDFlMDY2ODU2MTJjZjMxMDYyOThjMGI4MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RVFHS04KVla8xhyDk7xP91ln65DI36b9gQwQbzbkS10">
> And with these colors it's very hard to read

You're totally right I just changed it to be black text so it's more readable on all devices.

cocoStein reviewed (approved):
Thanks for the fix with the colors !
